evdi-kmod-common.spec

```diff
@@ -0,0 +1,94 @@
+Name:          evdi-kmod-common
+Version:       %{_version}
+Release:	1%{?dist}
+Summary:       Userland library for Extensible Virtual Display Interface Kernel module
+URL:           https://github.com/DisplayLink/evdi
+Source0:       %{url}/archive/v%{version}.tar.gz
+Source1:        displaylink.service
+Source2:        99-displaylink.rules
+Source3:        displaylink-sleep-extractor.sh
+# From http://www.displaylink.com/downloads/ubuntu.php
+Source4:	DisplayLink USB Graphics Software for Ubuntu %{_daemon_version}.zip
+Source5:        20-displaylink.conf
+License:       GPLv2.1, MIT and Proprietary
+ExclusiveArch:  i386 x86_64
+BuildRequires:  libdrm-devel
+BuildRequires:  make
+
+# No sources, so no debug package possible.
+%global debug_package %{nil}
+
+%description
+The evdi %{version} display driver userland libraries
+
+%prep
+%setup -q -c evdi-%{version}
+unzip "%{SOURCE4}"
+chmod +x displaylink-driver-%{_daemon_version}.run
+./displaylink-driver-%{_daemon_version}.run --noexec --keep
+
+%build
+pushd evdi-%{version}
+make -C library
+popd
+
+%install
+%define debug_package %{nil}
+mkdir -p $RPM_BUILD_ROOT/usr/libexec/displaylink/ \
+         $RPM_BUILD_ROOT/var/log/displaylink/
+
+# Library
+cp evdi-%{version}/library/libevdi.so $RPM_BUILD_ROOT/usr/libexec/displaylink
+
+# DisplayLinkManager
+%ifarch x86_64
+cp -a displaylink-driver-%{_daemon_version}/x64-ubuntu-1604/DisplayLinkManager $RPM_BUILD_ROOT/usr/libexec/displaylink/
+%endif
+
+%ifarch %ix86
+cp -a displaylink-driver-%{_daemon_version}/x86-ubuntu-1604/DisplayLinkManager $RPM_BUILD_ROOT/usr/libexec/displaylink/
+%endif
+
+# Firmwares
+cp -a displaylink-driver-%{_daemon_version}/ella-dock-release.spkg displaylink-driver-%{_daemon_version}/firefly-monitor-release.spkg $RPM_BUILD_ROOT/usr/libexec/displaylink/
+
+# systemd/udev
+mkdir -p $RPM_BUILD_ROOT/usr/lib/systemd/system/
+cp -a %{SOURCE1} $RPM_BUILD_ROOT/usr/lib/systemd/system/
+mkdir -p $RPM_BUILD_ROOT/etc/udev/rules.d/
+cp -a %{SOURCE2} $RPM_BUILD_ROOT/etc/udev/rules.d/
+mkdir -p $RPM_BUILD_ROOT/etc/X11/xorg.conf.d/
+cp -a %{SOURCE5} $RPM_BUILD_ROOT/etc/X11/xorg.conf.d/
+
+# pm-util
+mkdir -p $RPM_BUILD_ROOT/usr/lib/systemd/system-sleep/
+bash %{SOURCE3} displaylink-driver-%{_daemon_version}/displaylink-installer.sh > $RPM_BUILD_ROOT/usr/lib/systemd/system-sleep/displaylink.sh
+
+chmod +x $RPM_BUILD_ROOT/usr/lib/systemd/system-sleep/displaylink.sh
+
+%post
+%systemd_post %{SOURCE1}
+
+%files
+%doc LICENSE
+/usr/lib/systemd/system/displaylink.service
+/usr/lib/systemd/system-sleep/displaylink.sh
+/etc/udev/rules.d/99-displaylink.rules
+/etc/X11/xorg.conf.d/20-displaylink.conf
+%dir /usr/libexec/displaylink
+/usr/libexec/displaylink/*
```

Comment:
I think in this case, if it's possible, it would be nice to have an exception to leave it that way so that we have less problems with forward-compatibility in case displaylink renames or adds new files here, and we miss those on our packaging update (which we might not notice in a CI environment without manual verification).

My opinion here is that hardening against delivering an incomplete release is likely a better trade-off than the alternative, but please feel free to convince me otherwise :-)

Reply:
I understand that from a single package point of view, using wildcard seems convenient. But as a package maintainer, we want to ensure that no unexpected binaries will conflict with another package on end-user "before" they installed on their system.
This can be achieved by listing them explicitly.
So something like displaylink* is appropriate, but display* (it may conflicts with ImageMagick  package that has a display binary among others).

Reply:
In this case we are already inside `/usr/libexec/displaylink`, that's why I thought it would be *safe enough*. The case I can think of is that the user manually installed displaylink before installing the RPM, but that is an expected clash. Other packages should not be expected to need the `/usr/libexec/displaylink` folder, or should we not make that assumption?

Reply:
Using libexec/displaylink is a good idea, specially for content that isn't supposed to be used by end-users directly. Should we expect some binaries to be used by end-users ?
Is this directory relevant for all binaries ?